servant/src/Servant/API/ContentTypes.hs

```diff
@@ -307,6 +304,28 @@ instance MimeRender OctetStream ByteString where
 instance MimeRender OctetStream BS.ByteString where
     mimeRender _ = fromStrict
 
+-- | A type for responses with content-body.
+data NoContent = NoContent
+  deriving (Show, Eq, Read)
+
+instance FromJSON NoContent where
+    parseJSON _ = return NoContent
```

Comment:
Turns out `aeson` does crash with no content. However, there is an overlapping instance for `NoContent` in `servant-client`, in which we don't even try parsing the body. I don't know if that's good enough.